docs/rfcs/1012-parsec-auth.md

```diff
@@ -0,0 +1,430 @@
+
+<!-- Device stored on a third-party service -->
+```mermaid
+sequenceDiagram
+  autonumber
+  participant Client
+  participant S_Auth as Authentification server
+  participant S_Data as Metadata server
+
+  Client  ->> S_Auth: Retrieve the device of the user
+  S_Auth -->> Client: Device of the user
+
+  Client  ->> S_Data: Exchange with the server
+```
+
+### Account creation
+
+Creating the account for a user will require some information for the system to work:
+
+- An identifier for the user (email)
+
+  The service needs to verify that the user has access to the email (using a code or sending a link to continue the process).
+
+- A HMAC shared key to authenticate the user (`AUTH_MEDIUM_HMAC_KEY`)
+
+  When using a password, it's not used as is, but it's derived to create a HMAC key using the function `PBKDF_A` (this operation is done on the client side to not leak the master password).
+  The new secret will be provided as is to the server (like a password).
+
+  For the FIDO2 case the included challenge is used as is since the private key is not shared with the server.
+
+- An asymmetric key pair (`AUTH_MEDIUM_PUB_KEY`, `AUTH_MEDIUM_PRIV_KEY`)
+
+  They're generated using a cryptographic secure pseudo-random number generator (CSPRNG).
+
+  > This step is not needed when using a fido2 device since, it provides the asymmetric keys securely.
+
+- A symmetric key used to encrypt `AUTH_MEDIUM_PRIV_KEY` (`AUTH_MEDIUM_SYM_KEY`) alongside its parameters used to generate it `AUTH_MEDIUM_SYM_KEY_PARAMS`.
+
+  When using a password, it's generated from the function `PBKDF_B`
+
+  For FIDO2, we can bypass that step since we do not share `AUTH_MEDIUM_PRIV_KEY` since it's stored on the device.
+
+- A randomly generated symmetric key `ACCOUNT_MANIFEST_SYM_KEY`, it will be used to encrypt the account manifest.
+
+- Generate the account manifest `ACCOUNT_MANIFEST`,
+
+  the manifest will store:
+
+  - The list of authentication public keys
+  - The list of keys used to encrypt the devices (the list is empty at the beginning)
+
+  > The manifest will be encrypted to prevent the server to temper with it.
+
+> [!CAUTION]
+> The output of `PBKDF_A` and `PBKDF_B` should be different to avoid leaking the private key.
+
+
+```mermaid
+sequenceDiagram
+  autonumber
+  participant Alice
+  participant S as Authentification server
+
+  Alice ->> S: Provide its email for the account creation
+  S ->> Alice: Send an email with a unique code need to continue the process
+  alt Choose between password or fido2
+    Alice ->> Alice: Choose a password
+    Alice ->> Alice: Derive `AUTH_MEDIUM_HMAC_KEY` using the password and `PBKDF_A`
+    Alice ->> Alice: Create `AUTH_MEDIUM_SYM_KEY` using the password and `PBKDF_B`
+    Alice ->> Alice: Generate `AUTH_MEDIUM_PRIV_KEY`, `AUTH_MEDIUM_PUB_KEY`
```

Comment:
is there a real difference between create and generate ? 

Is it because the symetric key uses an input , and the asymetric pair doesn't ?